lib/Setup Thermostats.js

```diff
@@ -153,6 +155,7 @@ exports.setupThermostat = function (that, services)
 			.updateOnHubEvents(that.currentAccessory.id, "heatingSetpoint")	
 			.on('set', function(newHomekitValue, callback, context)
 				{
+					if(TargetState.value !== 3) return callback(null);

```

Comment:
According to Hubitat Staff (see post https://community.hubitat.com/t/thermostat-capability-and-attributes/54960/7) you are supposed to send these setpoint values even when in "Auto" mode. I don't think either of us have these quite right though. I think if ( TargetState.value == 3) the code then also needs to check (CurrentState.value) to determine if operating in heating or cooling mode and send only the relevant one of cooling or heating setpoint. Need to give this some more thought.

Reply:
Totally agree with that, could be TargetState.value || CurrentState.value.

Further digging into this: I was able to verify that when setting to "Heat" and a temperature in a scene in the Home app, HomeKit does set these thresholds as well (which we've discussed rejecting). These commands *can* be removed in some more advanced apps like Home+ that expose each call, but the majority of users aren't paying for more sophisticated HomeKit apps.

Reply:
Been giving this some additional thought.

One trick on this one of course will be the order in which these commands arrive if sent in sequence.

Stands to reason that TargetState **should** always be the last state sent, so ideally there's not a lot of difference between Target and Current. However if someone is in heat/cool and the TargetState is set to auto in a scene, then there's a possibility CurrentState could be misleading...

I'd definitely want to ensure, however that **TargetState** is set up to reflect **CurrentState** on startup and based on device update since MakerAPI doesn't surface TargetState.

Otherwise I'm not sure that CurrentState adds any value.